src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -228,4 +231,102 @@ describe('<SegmentControl />', () => {
       expect(container).toMatchSnapshot();
     });
   });
+
+  describe('subText', () => {
+    it('renders subText for outlined & medium segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="outlined" size="medium">
+          <SegmentControl.Option
+            value="Option 1"
+            testID="segment-option-1"
+            subText="Some sub text"
+          />
+          <SegmentControl.Option
+            value="Option 2"
+            testID="segment-option-2"
+            subText="Some sub text"
+          />
+          <SegmentControl.Option
+            value="Option 3"
+            testID="segment-option-3"
+            subText="Some sub text"
+          />
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders subText for filled & medium segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="filled" size="medium">
+          <SegmentControl.Option
+            value="Option 1"
+            testID="segment-option-1"
+            subText="Some sub text"
+          />
+          <SegmentControl.Option
+            value="Option 2"
+            testID="segment-option-2"
+            subText="Some sub text"
+          />
+          <SegmentControl.Option
+            value="Option 3"
+            testID="segment-option-3"
+            subText="Some sub text"
+          />
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('throws error for subText with small size filled variant', () => {
```

Comment:
I think unit tests are missing for Icon as well? My bad, noticed it now, for `size=small` Icon component should not exists.